### ISSUE TITLE
Wrong nullable types with fluent API in Prisma 5.0

### ISSUE THREAD

**Issue by p-kuen** [prisma/prisma#20183]
### Bug description

We use Fluent API very often and it works very well.
Since Prisma version 5.0, results for related models queried with fluent API are nullable, although `findUniqueOrThrow` was used on the base model and the relation field is not optional.

### How to reproduce

1. Create two models with a 1:n relation:
```prisma
model PlanGroup {
  id             Int                     @id @default(autoincrement()) @db.UnsignedInt
  plans          BuildingSitePlan[]
}

model Plan {
  id              Int         @id @default(autoincrement()) @db.UnsignedInt
  groupId         Int         @map("group_id") @db.UnsignedInt
  group           PlanGroup   @relation(fields: [groupId], references: [id], onDelete: Cascade)
}
```
3. Use Fluent API to query them:
```ts
(
    await prisma.plan
      .findUniqueOrThrow({
        where: {
          id,
        },
      })
      .group({
        select: {
          id: true,
        },
        
      })
  ).id
```
6. See typescript error ts(2531) "Object is possibly `null`"

### Expected behavior

In versions <5.0.0, queries for non-nullable related models with fluent api were non-nullable.

### Prisma information

```prisma
model PlanGroup {
  id             Int                     @id @default(autoincrement()) @db.UnsignedInt
  plans          BuildingSitePlan[]
}

model Plan {
  id              Int         @id @default(autoincrement()) @db.UnsignedInt
  groupId         Int         @map("group_id") @db.UnsignedInt
  group           PlanGroup   @relation(fields: [groupId], references: [id], onDelete: Cascade)
}
```

```ts
(
    await prisma.plan
      .findUniqueOrThrow({
        where: {
          id,
        },
      })
      .group({
        select: {
          id: true,
        },
        
      })
  ).id
```

### Environment & setup

- OS: macOS Ventura 13.2
- Database: MariaDB 10.11.3
- Node.js version: v20.2.0

### Prisma Version

```
prisma                  : 5.0.0
@prisma/client          : 5.0.0
Current platform        : darwin-arm64
Query Engine (Node-API) : libquery-engine 6b0aef69b7cdfc787f822ecd7cdc76d5f1991584 (at node_modules/.pnpm/@prisma+engines@5.0.0/node_modules/@prisma/engines/libquery_engine-darwin-arm64.dylib.node)
Schema Engine           : schema-engine-cli 6b0aef69b7cdfc787f822ecd7cdc76d5f1991584 (at node_modules/.pnpm/@prisma+engines@5.0.0/node_modules/@prisma/engines/schema-engine-darwin-arm64)
Schema Wasm             : @prisma/prisma-schema-wasm 4.17.0-26.6b0aef69b7cdfc787f822ecd7cdc76d5f1991584
Default Engines Hash    : 6b0aef69b7cdfc787f822ecd7cdc76d5f1991584
Studio                  : 0.487.0
```

**Comment by Jolg42** [comment]
@p-kuen and upvoters: The fix is merged and will be a part of 5.1 release.
You can try `5.1.0-dev.28` development snapshot and let us know if that version works for you. We don't recommend using it in production, but it should be enough to confirm the fix.

**Comment by p-kuen** [comment]
Thanks for the fix! The PR shows a negative impact in typescript compilation, do I read this correctly? But I think it's still better than in versions before 5.0.0, right?

**Comment by Jolg42** [comment]
@p-kuen I guess you saw the "Codspeed" comment https://github.com/prisma/prisma/pull/20197#issuecomment-1633110739 It is unfortunately not reliable for this, so it's hard to say if there is any actual performance impact.
I definitely recommend ignoring it and using 5.0.0 and up.

**Comment by p-kuen** [comment]
I can confirm that the Fluent API error is fixed with current dev version (5.1.0-dev.61), although I have lots of other errors with enums and other typings. We will wait until the release of 5.1.0 and look if problems still exist.
Thanks again for the fix!
